Allow maximum published package size to be configured when using IIS

RichardSinden opened the pull request:
Now sets the MaxRequestBodySize for IIS on startup.

 * I've modified the startup code to set the IISServerOptions based on configuration in the appsettings.json file, with a default entry for MaxRequestBodySize with a maximum size of 250MB.
 * I've updated the documentation

Addresses https://github.com/loic-sharma/BaGet/issues/463

loic-sharma commented on src/BaGet/Extensions/IServiceCollectionExtensions.cs:
```diff
@@ -191,6 +193,15 @@ public static IServiceCollection AddBaGetContext(this IServiceCollection service
+            this IServiceCollection services,
+            IConfiguration configuration)
+        {
+            services.ConfigureAndValidate<IISServerOptions>(configuration.GetSection(nameof(IISServerOptions)));
```
I'd like to keep the appsettings.json file as light as possible. What do you think of setting the default of 250MB here? I think this should work:

```csharp
// IIS defaults upload sizes to ~30MB. Increase it to 250MB.
// See https://github.com/aspnet/Announcements/issues/267
services.Configure<IISServerOptions>(iis=>
{
    iis.MaxRequestBodySize = 262144000
});

services.ConfigureAndValidate<IISServerOptions>(configuration.GetSection(nameof(IISServerOptions)));
```

I believe this should default to 250MB, yet let customers override that value by adding a `IISServerOptions` option.

loic-sharma commented on src/BaGet/appsettings.json:
```diff
@@ -22,6 +22,10 @@
 
+  "IISServerOptions": {
+    "MaxRequestBodySize": 262144000
+  },
```
If you go with my comment above https://github.com/loic-sharma/BaGet/pull/473/files#r378645236, please remove these lines since they are no longer necessary for the 250MB default.

loic-sharma commented:
Thank you for figuring this problem out. I'm a bit of an IIS noob so I really appreciate the help! :)

RichardSinden commented:
Yep, I agree, I wasn't entirely comfortable with always having config for IIS in the appsettings even when you are not using IIS. Your solution is better and I'll update it when I get the chance later, although I was going to wait and see if there was more information on this comment https://github.com/loic-sharma/BaGet/issues/463#issuecomment-585573492, in case we don't need to do anything at all.

softlion commented:
Is this MaxRequestBodySize not enough ?

```
  .ConfigureKestrel((context, options) =>
    {
        options.Limits.MaxRequestBodySize = ...;
    });
```

RichardSinden commented:
> 
> 
> Is this MaxRequestBodySize not enough ?
> 
No, this setting doesn't seem to have any effect when hosting in IIS. 

@loic-sharma, I've updated the PR as suggested to set a default in code and remove the appsettings.json default section. I've updated the documentation to make reference to the IISServerOptions section in the Microsoft docs.

loic-sharma commented:
Thanks for the contribution! :)
